Remotion/Data/DomainObjects/Mapping/Builder/ClassDefinitionBuilderNode.cs

```diff
@@ -25,32 +27,55 @@ namespace Remotion.Data.DomainObjects.Mapping.Builder
   /// Represents <see cref="Mapping.ClassDefinition"/> node in a tree constructed by <see cref="TypeDefinitionFactory"/>.
   /// Supports child navigation and correctly setting up child relationships in a resulting <see cref="Mapping.ClassDefinition"/>.
   /// </summary>
+  [DebuggerDisplay("Builder node for class '{Type}'")]
   public class ClassDefinitionBuilderNode : TypeDefinitionBuilderNode
   {
-    public ClassDefinitionBuilderNode? BaseClass { get; }
+    public ClassDefinitionBuilderNode? BaseClassNode { get; }
+
+    public IReadOnlyList<InterfaceDefinitionBuilderNode> ImplementedInterfaceNodes { get; }
+
+    [MemberNotNullWhen(false, nameof(BaseClassNode))]
+    public bool IsInheritanceRoot { get; }
 
     public ClassDefinition? ClassDefinition { get; private set; }
 
     public bool IsConstructed { get; private set; }
 
-    private readonly List<ClassDefinitionBuilderNode> _derivedClasses = new();
+    private readonly List<ClassDefinitionBuilderNode> _derivedClassNodes = new();
 
-    public ClassDefinitionBuilderNode (Type type, ClassDefinitionBuilderNode? baseClass)
+    public ClassDefinitionBuilderNode (Type type, ClassDefinitionBuilderNode? baseClassNode, IEnumerable<InterfaceDefinitionBuilderNode> implementedInterfaceNodes)
         : base(type)
     {
+      ArgumentUtility.CheckNotNull("implementedInterfaceNodes", implementedInterfaceNodes);
       if (!type.IsClass)
         throw new ArgumentException("The specified type must be a class.", "type");
-      if (baseClass?.IsConstructed == true)
-        throw new ArgumentException("The specified base class must not be a constructed node.", "baseClass");
-
-      BaseClass = baseClass;
-      BaseClass?.AddDerivedClass(this);
+      if (type == typeof(DomainObject))
+        throw new ArgumentException("Cannot create a builder node for the DomainObject type.", "type");
+      if (!ReflectionUtility.IsDomainObject(type))
+        throw new ArgumentException("Cannot create a builder node for a type that is not a domain object.", "type");
```

Comment:
message: in arg checks, the behavior is usually not listed, the message focuses on the value itself. this reads more like an invalidopex.
 "specified type must not be DomainObject"
"specified  type must be derived from DomainObject"